panoramapublic/src/org/labkey/panoramapublic/query/SubmissionManager.java

```diff
@@ -0,0 +1,413 @@
+package org.labkey.panoramapublic.query;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.labkey.api.data.CompareType;
+import org.labkey.api.data.CoreSchema;
+import org.labkey.api.data.DbScope;
+import org.labkey.api.data.SQLFragment;
+import org.labkey.api.data.SimpleFilter;
+import org.labkey.api.data.Sort;
+import org.labkey.api.data.SqlExecutor;
+import org.labkey.api.data.Table;
+import org.labkey.api.data.TableSelector;
+import org.labkey.api.portal.ProjectUrls;
+import org.labkey.api.query.FieldKey;
+import org.labkey.api.query.ValidationException;
+import org.labkey.api.security.MutableSecurityPolicy;
+import org.labkey.api.security.SecurityPolicyManager;
+import org.labkey.api.security.User;
+import org.labkey.api.security.permissions.UpdatePermission;
+import org.labkey.api.security.roles.EditorRole;
+import org.labkey.api.util.PageFlowUtil;
+import org.labkey.api.view.ActionURL;
+import org.labkey.api.view.ShortURLRecord;
+import org.labkey.api.view.ShortURLService;
+import org.labkey.panoramapublic.PanoramaPublicController;
+import org.labkey.panoramapublic.PanoramaPublicManager;
+import org.labkey.panoramapublic.model.DataLicense;
+import org.labkey.panoramapublic.model.ExperimentAnnotations;
+import org.labkey.panoramapublic.model.Journal;
+import org.labkey.panoramapublic.model.JournalExperiment;
+import org.labkey.panoramapublic.model.JournalSubmission;
+import org.labkey.panoramapublic.model.Submission;
+
+import java.util.Comparator;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class SubmissionManager
+{
+    private static final Logger LOG = LogManager.getLogger(SubmissionManager.class);
+
+    public static Submission getSubmission(int id)
```

Comment:
I worry that many of these methods that take an ID are subject to direct object reference. I spot checked a few of them and it looks like the callers are not validating the owning container. So if you can guess the ID for a row you don't have access to (not that hard since they're auto-incrementing) you may be able to view or manipulate it.

To fix it, you'd pass in a Container (if you know the container it'll be in), or a User and make sure that the user has at least read permission in the object's container.

https://cheatsheetseries.owasp.org/cheatsheets/Insecure_Direct_Object_Reference_Prevention_Cheat_Sheet.html

Reply:
Actions that operate on a submission request need a valid ExperimentAnnotations object.  And all of the relevant action classes in PanoramaPublicController call the ensureCorrectContainer(Container requestContainer, Container expAnnotContainer, ViewContext viewContext) method that throws a RedirectException if the container in the request is not the same as the owning container of the ExperimentAnnotations object. 
So it would not be possible for a user to edit/update/delete a submission request by using the row Id for a submission that belonged to an experiment in a container where they do not have the required permission. 

But I've also added a Container parameter to SubmissionManager methods that take a database row ID, as you suggested.